freedomofpress/securedrop
# Update dev container browser software and validation

## Status

Ready for review

## Description of Changes

[Mozilla's keyserver went down, and they might just leave it that way.](https://github.com/mozilla/gpg.mozilla.org/issues/7) Their release signing key is available on another one, but that doesn't seem to be the result of any official policy I can find. The closest thing to guidance comes from a [blog post](https://blog.mozilla.org/security/2019/06/13/updated-firefox-gpg-key/) instructing us to obtain the key from the archive server along with the software. This changes the Dockerfile to do that.

Recent versions of geckodriver are being signed with the release key, so we'll now use it to verify that package too.

We were also using an expired version of the Tor release signing key, kept in our repo, so I'm following their [instructions](https://support.torproject.org/tbb/how-to-verify-signature/) for obtaining the key instead.

Finally, this updates all three packages.

Finally finally, this includes a workaround for the recent `setuptools` breakage.

Fixes #5470.
Fixes #5471.

## Testing

- Check out the `5470-keystone-kops` branch. 
- Start the dev server with `make dev`. Confirm that the development container is built and started properly.

## Deployment

Dev/test only.

## Checklist

### If you made changes to the server application code:


### If you made non-trivial code changes:
